### ISSUE TITLE
'ReLU' object has no attribute 'y'

### ISSUE THREAD

**Issue by florianst** [shap/shap#728]
With sequential pytorch layers comprising a ReLU, such as this one
```
        self.convLayer1 = nn.Sequential(
            nn.Conv1d(256, 512, kernel_size=2, stride=1, padding=0),
            nn.ReLU()
            )
```
I run into the following error:
```
Traceback (most recent call last):
  File "train.py", line 420, in <module>
    shap_values = explainer.shap_values(to_explain)
  File "/users/florianst/.local/lib/python3.6/site-packages/shap-0.29.3.dev0-py3.6-linux-x86_64.egg/shap/explainers/deep/__init__.py", line 119, in shap_values
    return self.explainer.shap_values(X, ranked_outputs, output_rank_order)
  File "/users/florianst/.local/lib/python3.6/site-packages/shap-0.29.3.dev0-py3.6-linux-x86_64.egg/shap/explainers/deep/deep_pytorch.py", line 172, in shap_values
    sample_phis = self.gradient(feature_ind, joint_x)
  File "/users/florianst/.local/lib/python3.6/site-packages/shap-0.29.3.dev0-py3.6-linux-x86_64.egg/shap/explainers/deep/deep_pytorch.py", line 115, in gradient
    for idx, x in enumerate(X)]
  File "/users/florianst/.local/lib/python3.6/site-packages/shap-0.29.3.dev0-py3.6-linux-x86_64.egg/shap/explainers/deep/deep_pytorch.py", line 115, in <listcomp>
    for idx, x in enumerate(X)]
  File "/users/florianst/.local/lib/python3.6/site-packages/torch/autograd/__init__.py", line 149, in grad
    inputs, allow_unused)
  File "/users/florianst/.local/lib/python3.6/site-packages/shap-0.29.3.dev0-py3.6-linux-x86_64.egg/shap/explainers/deep/deep_pytorch.py", line 213, in deeplift_grad
    return op_handler[module_type](module, grad_input, grad_output)
  File "/users/florianst/.local/lib/python3.6/site-packages/shap-0.29.3.dev0-py3.6-linux-x86_64.egg/shap/explainers/deep/deep_pytorch.py", line 337, in nonlinear_1d
    delta_out = module.y[: int(module.y.shape[0] / 2)] - module.y[int(module.y.shape[0] / 2):]
  File "/users/florianst/.local/lib/python3.6/site-packages/torch/nn/modules/module.py", line 539, in __getattr__
    type(self).__name__, name))
AttributeError: 'ReLU' object has no attribute 'y'
```
I'm running the current master branch.

**Comment by gabrieltseng** [comment]
Hi @florianst ,

The problem is that the `y` and `x` attributes are deleted - this was done before I'd considered a use case for multiple inputs, and didn't think those attributes would need to be reused.

I'll open a PR to fix this, but in the meantime deleting lines 326, 327, 348, 349 from the [pytorch deep explainer](https://github.com/slundberg/shap/blob/master/shap/explainers/deep/deep_pytorch.py) should fix it.

Alternatively, it's now working in [my fork](https://github.com/gabrieltseng/shap/tree/pytorch/improvements) of shap.

Sorry about that!

**Comment by florianst** [comment]
Fixed in #733
